components/builder-web/app/shared/pipes/keys.pipe.ts

```diff
@@ -0,0 +1,8 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'habKeysPipe', pure: false })
+export class KeysPipe implements PipeTransform {
+  transform(value: any, args: any[] = null): any {
+    return Object.keys(value);
```

Comment:
This feels like more than what's needed, and makes the templates a bit confusing. If you want just the keys, can't you just ask the component for `integrationTypes` and have that return `Object.keys`?

Reply:
I thought about that and I think I probably went for the YAGNI approach. I used a pipe like this quite a bit in another unnamed project so I thought it might be useful. I'm amenable to changing it if you think we won't need it.

Reply:
Ok cool, yeah. I guess if you find you need jus-the-keys that often, it's likely a smell that you could use a structure to store them that way (e.g., a property on the state that's just a list of those keys).